src/control/server/harness.go

```diff
@@ -186,17 +186,13 @@ func (h *IOServerHarness) registerNewMember(membership *system.Membership, insta
 		return errors.Wrap(err, "failed to get member from instance")
 	}
 
-	created, oldState, err := membership.AddOrUpdate(m)
-	if err != nil {
-		return errors.Wrap(err, "failed to add MS replica to membership")
-	}
-
+	created, oldState := membership.AddOrUpdate(m)
 	if created {
 		h.log.Debugf("bootstrapping system member: rank %d, addr %s\n",
 			m.Rank, m.Addr)
 	} else {
 		h.log.Debugf("updated bootstrapping system member: rank %d, addr %s, %s->%s\n",
-			m.Rank, m.Addr, *oldState, m.State)
+			m.Rank, m.Addr, *oldState, m.State())
```

Comment:
Strictly speaking, there is a race here which could result in inaccurate reporting on the results of the AddOrUpdate() operation. I don't think it matters in this case because it's only being used for a debug log statement, but I wanted to point it out.

If there is eventually logic that depends on an accurate representation of the state transition, then it would be better to return the current state from AddOrUpdate() to ensure that it was read under the same lock that was used to set it.